### PR TITLE
Nomcopter/yerna fixes

### PR DESCRIPTION
**Pull request by nomcopter**
@seansfkelley 

#33 #24 #32 #29

@gogwilt

### CODE REVIEW

File: lib/yarn-with-package-json-mangling.js [hunk 1]
```diff
@@ -18,7 +18,15 @@ function alphabetizeKeys(object) {
 
 function manglePackageJson(packageJsonPath, packagesByName) {
   const packageJson = fs.readFileSync(packageJsonPath);
-  const mangledPackageJson = JSON.parse(packageJson);
+
+  let mangledPackageJson;
+  try {
+    mangledPackageJson = JSON.parse(packageJson);
+  } catch (e) {
+    console.error('yerna: Could not parse ' + packageJsonPath);
```
Comment (seansfkelley) [inline lib/yarn-with-package-json-mangling.js]: Use `logger.error` here.

File: lib/taskrunning.js [hunk 2]
```diff
@@ -54,13 +54,14 @@ function createTaskRunner(spawnChild, ignoreFailures = false) {
...
-            mergedOutput.forEach(({ level, loglines }) => loglines.forEach(l => logger[level](l)));
-          } else {
-            logger.info(`yerna: completed ${chalk.cyan(pkg.name)} with no output (${incrementAndGetCompletedPackageCount()}/${getTotalPackageCount()})`);
+        if (mergedOutput.length) {
+          logger.info(`yerna: output for ${chalk.cyan(pkg.name)} (${incrementAndGetCompletedPackageCount()}/${getTotalPackageCount()})`);
+          mergedOutput.forEach(({ level, loglines }) => loglines.forEach(l => logger[level](l)));
+          if (isAborting) {
+            logger.info('-- ABORTED --');
           }
+        } else if (!isAborting) {
```
Comment (seansfkelley) [inline lib/taskrunning.js]: Does it make sense to do this check anymore? It seems like the collected output should be unconditionally forwarded, and then the `-- ABORTED --` thing printed at the end if appropriate. Maybe a phrasing change around "completed without output" to in the case that it was aborted to "aborted without output" so it doesn't look like it completed successfully?

Also, for scanning/filtering purposes, it'd be handy to prefix the aborted message with `yerna:`.

File: bin/yerna [hunk 3]
```diff
@@ -183,6 +187,7 @@ commander.on('--help', () =>{
     '    $ yerna install -- --pure-lockfile',
     '    $ yerna install --include foo --include bar -- --pure-lockfile',
     '    $ yerna run taskname --include foo --exclude bar',
+    '    $ yerna run taskname -i ^foo(bar)? --mangle',
```
Comment (seansfkelley) [inline bin/yerna]: Does this example actually run in a shell with ^ and ? without needing quoting?

File: bin/yerna [hunk 4]
```diff
@@ -111,13 +111,17 @@ const performList = decorate(function() {
...
 }, WithAbort, WithInit);
 
-const performRun = decorate(function(scriptName, moreArgs) {
+const performRun = decorate(function(scriptName, moreArgs, options) {
   const filteredPackages = filterPackages(PACKAGES, commander, pkg => !!pkg.scripts[scriptName]);
   const yarnArgs = [ 'run', scriptName  ].concat(moreArgs);
-  logFlagFeedback(commander, filteredPackages, 'yarn ' + yarnArgs.join(' '), [ `have a ${chalk.magenta(scriptName)} script` ]);
+  const executableName = options.mangle ? 'yarnhack' : 'yarn';
+  logFlagFeedback(commander, filteredPackages, executableName + ' ' + yarnArgs.join(' '), [ `have a ${chalk.magenta(scriptName)} script` ]);
+  const task = options.mangle
```
Comment (seansfkelley) [inline bin/yerna]: Nit: `spawnTask` or something verby.

File: bin/yerna (hunk 3 above)
Comment (nomcopter) [reply bin/yerna]: Nope! :awesome:

**Comment by nomcopter** [comment]
Correct!

**Pull request closed without merging**